src/rust/engine/process_execution/src/remote.rs

```diff
@@ -126,7 +128,7 @@ impl super::CommandRunner for CommandRunner {
   ///
   /// TODO: Request jdk_home be created if set.
   ///
-  fn run(&self, req: ExecuteProcessRequest) -> BoxFuture<FallibleExecuteProcessResult, String> {
+  fn run(&self, req: ExecuteProcessRequest, workunit_store: WorkUnitStore) -> BoxFuture<FallibleExecuteProcessResult, String> {
```

Comment:
@illicitonion : Should this be using thread/task locals like logging does? AFAIK, most tracing implementations do.

Reply:
I'm personally a big fan of explicit over implicit, but it could if we wanted to...

Reply:
Well, tracing is a lot like logging. Doing the equivalent thing there and propagating our logger explicitly throughout our callstack to all the places we might want to log would probably be too much.

Maybe this is not similar to a logger, and so that doesn't make sense here... unknown.

Reply:
As I use task locals in not many places in this PR I would prefer to leave it as it is. 
But this is a useful suggestion and it would be great to reconsider it when adding right parent_id  to all other v2 Nodes.